extract_endpoint/src/extract_endpoint/endpoint.py

```diff
@@ -126,28 +158,23 @@ def upload_file() -> typing.Tuple[str, int]:
         flask.abort(HTTPStatus.BAD_REQUEST)
 
     try:
-        file_z = zipfile.ZipFile(file)
+        zipfile.ZipFile(file)
+        file.seek(0)
     except zipfile.BadZipFile:
         LOGGER.warning("Bad zipfile")
         return "Bad Zipfile", HTTPStatus.BAD_REQUEST
 
-    for json_file in [file_z.open(zipinfo) for zipinfo in file_z.infolist()]:
-        if not azure_utils.upload_bytes_to_azure(App.config['AZURE_COORDINATES'], json_file.read(),
-                                                 utils.secure_filename(json_file.name)):
-            LOGGER.warning("File upload to Azure storage failed")
-            flask.abort(HTTPStatus.BAD_GATEWAY)
-    LOGGER.info(f"{len(file_z.infolist())} json files uploaded to Azure")
+    if ThreadRunner.is_thread_running():
+        LOGGER.warning("Still uploading to Azure, ignoring current upload")
+        flask.abort(HTTPStatus.TOO_MANY_REQUESTS)
 
     timestamp = flask.request.form['timestamp']
     if not azure_utils.upload_bytes_to_azure(App.config['AZURE_COORDINATES'], timestamp.encode(), TIMESTAMP_FILENAME):
         LOGGER.warning("Timestamp upload to Azure storage failed")
         flask.abort(HTTPStatus.BAD_GATEWAY)
 
-    run_tl_return_code = run_tl()
-    LOGGER.info(f"TL returned {run_tl_return_code}")
-    if run_tl_return_code in [HTTPStatus.OK, HTTPStatus.TOO_MANY_REQUESTS]:
-        return '', run_tl_return_code
-    return '', HTTPStatus.BAD_GATEWAY
+    ThreadRunner.start_new_thread(target=functools.partial(unzip_upload_run_tl, data=file.read()))
```

Comment:
python closes files when they go to a thread - why? :(  As a work around, pass the file contents and recreate the Zipfile in the thread.

Reply:
Annoying, but I guess that makes sense